itou/siaes/management/commands/import_siae_2020_04_13.py

```diff
@@ -0,0 +1,420 @@
+    assert len(rows) <= 1
+    return rows[0] if len(rows) == 1 else None
+
+
+def get_main_df_row_as_dict(external_id):
+    return get_df_row_as_dict(MAIN_DF, external_id)
+
+
+def get_secondary_df_row_as_dict(external_id):
+    return get_df_row_as_dict(SECONDARY_DF, external_id)
+
+
+def get_secondary_df_rows_as_dict(external_id):
+    return get_df_rows_as_dict(SECONDARY_DF, external_id)
+
+
+def should_siae_be_created(siae):
+    if siae.kind == Siae.KIND_ETTI:
+        return True
+    return siae.department in [
+        "62",
+        "67",
+        "75",
+        "77",
+        "78",
+        "91",
+        "92",
+        "93",
+        "94",
+        "95",
+        "08",
+        "10",
+        "51",
+        "52",
+        "54",
+        "55",
+        "57",
+        "67",
+        "68",
+        "88",
+    ]
+
+
+class Command(BaseCommand):
+    """
+    Update and sync SIAE data based on latest ASP exports.
+
+    To debug:
+        django-admin import_siae_2020_04_13 --verbosity=2 --dry-run
+
+    When ready:
+        django-admin import_siae_2020_04_13 --verbosity=2
+    """
+
+    help = "Update and sync SIAE data based on latest ASP exports."
+
+    def add_arguments(self, parser):
+        parser.add_argument("--dry-run", dest="dry_run", action="store_true", help="Only print data to import")
+
+    def set_logger(self, verbosity):
+        """
+        Set logger level based on the verbosity option.
+        """
+        handler = logging.StreamHandler(self.stdout)
+
+        self.logger = logging.getLogger(__name__)
+        self.logger.propagate = False
+        self.logger.addHandler(handler)
+
+        self.logger.setLevel(logging.INFO)
+        if verbosity > 1:
+            self.logger.setLevel(logging.DEBUG)
+
+    def log(self, message):
+        self.logger.debug(message)
+
+    def update_existing_siaes(self, dry_run):
+        for siae in Siae.objects.all():
+            if siae.external_id:
+                row = get_main_df_row_as_dict(external_id=siae.external_id)
+                if row:
+                    if row["siret"] == siae.siret:
+                        pass
+                    else:
+                        assert siae.siret[:9] == row["siret"][:9]
+                        assert siae.kind in EXPECTED_KINDS
+                        assert siae.source == Siae.SOURCE_ASP
+                        assert siae.members.count() == 0
+                        self.log(
+                            f"siae.id={siae.id} has changed siret from "
+                            f"{siae.siret} to {row['siret']} (will be updated)"
+                        )
+                        if not dry_run:
+                            siae.siret = row["siret"]
+                            siae.save()
+                    # FIXME update other fields as well. Or not?
+                    # Tricky decision since our users may have updated their data
+                    # themselves and we have no record of that.
```

Comment:
Tu fais un `assert siae.members.count() == 0` juste au dessus, comment un utilisateur aurait pu mettre à jour les données ? :p

Reply:
Non le `assert siae.members.count() == 0` est dans le bloc du `else`. Ce commentaire est indenté un cran avant. A terme ce script évoluera pour MAJ les SIAE existantes (nom, adresse...) ce que je ne fais pas dans cette itération car
- le focus est sur l'ajout de nouvelles SIAE
- j'aurai du mal à synchroniser ces changements de toute façon car cela nécéssiterait de rajouter des champs en db pour conserver les noms, adresses des SIAE originaux (ASP) avant que ceux-ci soient éventuellement modifiés par l'utilisateur SIAE. D'où mon commentaire.